### DISCUSSION TITLE
Directory Server traffic statistics

### DISCUSSION THREAD

**Discussion by corrados** [jamulussoftware/jamulus#3047] [General]
I just updated the Jamulus Directory server (Any Genre) with the newest security updates. Out of curiosity, I also checked the traffic which is logged by my provider:
![grafik](https://user-images.githubusercontent.com/46655886/232302314-7e2917d6-3bc9-45bd-afc8-63a75801a039.png)
It's interesting to notice that the traffic goes down almost linearly with time. Unfortunately, I do not have statistics available from last year. Would be interesting to see how it varied during the other months.

Other (related) topic: "Any Genre 3" only has a few registered servers. Maybe you should think about removing it from Jamulus? It does not hurt to keep it as is but some clean up would also not be a bad idea ;-).

**Comment by pljones** [comment]
> Maybe you should think about removing it from Jamulus?

I certainly think it wouldn't hurt to make it easier to have a more dynamic system of "core" directories (i.e. those we term the public ones).

**Reply by ann0see** [reply]
Suggestion: Read a config file containing  all directories. Instead of using hardcoded directories. Get rid of the concept of inbuilt ones. 

Issues:
* How to handle update checking
* Where to store the config name
* How should the UI managing the config file look like - or shouldn't there be any at all?

**Reply by pljones** [reply]
1. How to handle update checking
We can have the two update servers in the same config file - but we already use CNAME freely, so I don't think this is much of an issue.
2. Where to store the config name
Jamulus could ask the update server for the current version as part of its check and then store that until the update server next says "Here's a new set of public Directories".  That lets us implement the feature smoothly, too. [1]
3. How should the UI managing the config file look like - or shouldn't there be any at all?
I don't see this as "user managed" any more than the current "core" Directories can be.  The custom directories in Jamulus Client and Server would remain unchanged.

---
[1] This _also_ gets people running "off the net" Jamulus a way of hiding the core Directories - CNAME (or otherwise override) the update servers, get them to return only your own private custom Directory.  Then none of your users have to wonder "which directory should I use?"
